### PR TITLE
Basic gage info from NWIS

### PR DESCRIPTION
**Pull request by lindsaycarr**
Part of #3 (still need NWS flood info). I moved some of `1_fetch.yml` around because this step depends on the spatial steps, and the streamflow fetch depends on this.

For now, I just used `sample_n` to grab only 50 of the 3000+ sites. We need to have a greater discussion about how we are filtering sites down.

### CODE REVIEW

File: 1_fetch/src/fetch_sites.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  state_cds <- unique(dataRetrieval::stateCdLookup(state_names))
+
+  # Cast wide net for all NWIS sites with stage data that fall within that bbox
+  sites_df <- data.frame()
+  for(cd in state_cds) {
+    sites_df_cd <- dataRetrieval::whatNWISdata(stateCd = cd, parameterCd = stream_params$stage, service = "uv")
+    sites_df <- rbind(sites_df, sites_df_cd)
+  }
+  sites_df <- dplyr::filter(sites_df, site_tp_cd == "ST")
+  sites_df <- dplyr::filter(sites_df, end_date >= as.Date(dates$dates$end))
```
Comment (jzwart) [inline 1_fetch/src/fetch_sites.R]: I had to do this too, but any idea why we have to do `dates$dates$...` when we've already subset to `dates` in the `viz_config`?

File: 1_fetch/cfg/stream_params.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 flow:
   - '00060'
-  - '00072'
+stage:
+  - '00065'
```
Comment (jzwart) [inline 1_fetch/cfg/stream_params.yml]: need to change code in `fetch_streamflow` if we want stage + discharge

File: 1_fetch/src/fetch_sites.R [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+#' Subset the site list to ones relevant for the storm
+#'
+#' @param ind_file character file name where the output should be saved
+#' @param all_sites data.frame of sites pulled for the bbox
+subset_sites <- function(ind_file, all_sites) {
+
+  scipiper::sc_retrieve(all_sites)
+  all_sites <- readRDS(scipiper::as_data_file(all_sites))
```
Comment (jzwart) [inline 1_fetch/src/fetch_sites.R]: I think you can combine these lines to: `all_sties <- readRDS(sc_retrieve(all_sites))`

File: 1_fetch/src/fetch_streamdata.R [hunk 4]
```diff
@@ -0,0 +1,14 @@
+fetch_streamdata <- function(ind_file, sites, dates, stream_params, gd_config) {
+
+  scipiper::sc_retrieve(sites)
+  sites <- feather::read_feather(scipiper::as_data_file(sites))
+
+  message('  starting download of NWIS data at ', Sys.time())
+  data <- readNWISuv(
+    siteNumbers=sites$site_no, parameterCd=stream_params$stage,
```
Comment (jzwart) [inline 1_fetch/src/fetch_streamdata.R]: ah you got it, disregard my comment above

File: 1_fetch/src/fetch_sites.R (hunk 1 above)
Comment (lindsaycarr) [reply 1_fetch/src/fetch_sites.R]: I think @aappling-usgs fixed that here, but I haven't had a chance to merge those changes in just yet! https://github.com/USGS-VIZLAB/vizstorm-GIF/pull/24/commits/a58c5513b8f2039b5d5159d93fba36a36cbbca08

File: 1_fetch/src/fetch_sites.R (hunk 3 above)
Comment (lindsaycarr) [reply 1_fetch/src/fetch_sites.R]: 👍

**Pull request merged**